core/controllers/admin.py

```diff
@@ -656,6 +658,453 @@ def _generate_dummy_explorations(
+
+        Args:
+            num_sample_ops: int. Number of sample opportunities to
+                generate.
+            num_sample_interactions_per_opportunity: int. Number of sample
+                interactions to generate per opportunity.
+            should_submit_suggestions: boolean.
+                Track whether suggestions should be submitted.
+        """
+        topic_id = topic_fetchers.get_new_topic_id()
+        story_id = story_services.get_new_story_id()
+
+        topic = topic_domain.Topic.create_default_topic(
+            topic_id, 'Sample Topic 1', 'sample-topic-one', 'description')
+
+        topic.add_canonical_story(story_id)
+        topic.add_subtopic(1, 'Sample Subtopic Title')
+
+        subtopic_page = (
+        subtopic_page_domain.SubtopicPage.create_default_subtopic_page(
+            1, topic_id))
+
+        story = story_domain.Story.create_default_story(
+            story_id, 'Sample Story Title', 'Description',
+            topic_id, 'sample-story-title')
+
+        story_node_dicts = self._generate_sample_chapters(
+            num_sample_ops,
+            num_sample_interactions_per_opportunity, 0)
+
+        len_story = len(story_node_dicts)
+        for i, story_node_dict in enumerate(story_node_dicts, 1):
+            self._generate_sample_story_node(
+                story, len_story, i, **story_node_dict)
+
+        story_services.save_new_story(self.user_id, story)
+        topic_services.save_new_topic(self.user_id, topic)
+
+        subtopic_page_services.save_subtopic_page(
+            self.user_id, subtopic_page, 'Added subtopic',
+            [topic_domain.TopicChange({
+                'cmd': topic_domain.CMD_ADD_SUBTOPIC,
+                'subtopic_id': 1,
+                'title': 'Sample Subtopic Title'
+            })]
+        )
+
+        exp_ids_in_story = story.story_contents.get_all_linked_exp_ids()
+        opportunity_services.add_new_exploration_opportunities(
+            story_id, exp_ids_in_story)
+
+        if should_submit_suggestions:
+            self._submit_suggestions(
+                num_sample_ops,
+                num_sample_interactions_per_opportunity,
+                exp_ids_in_story)
+
+    def _generate_sample_opportunities_with_existing_topic(
+            self, num_sample_ops,
+            num_sample_interactions_per_opportunity,
+            should_submit_suggestions, story_url_fragment):
+        """Creates the given number of sample opportunities
+        and adds them to an existing story.
+
+        Args:
+            num_sample_ops: int. Number of sample opportunities to
+                generate.
+            num_sample_interactions_per_opportunity: int. Number of sample
+                interactions to generate per opportunity.
+            should_submit_suggestions: boolean.
+                Track whether suggestions should be submitted.
+            story_url_fragment: stringType. Story fragment of existing story.
```

Comment:
*string ?

Also, what does "Story fragment of existing story" mean? The description should something like "The url fragment of the story." right? And if the `story_url_fragment` is always set to 'sample-story-title', then it doesn't seem like this arg is necessary.